internal/convenience/decoder/decoder.go

```diff
@@ -73,7 +74,23 @@ func (o *OutputDecoder) HandleOutputV2(
 		"inputIndex", processOutputData.InputIndex,
 		"outputIndex", processOutputData.OutputIndex,
 	)
-	if processOutputData.Payload[2:10] == model.VOUCHER_SELECTOR {
+
+	input := &model.InputEdge{
+		Node: struct {
+			Index int    `json:"index"`
+			Blob  string `json:"blob"`
+		}{
+			Blob: processOutputData.Payload,
+		},
+	}
+	convertedInput, err := o.GetConvertedInput(*input)
+	if err != nil {
+		slog.Error("Failed to get converted:", "err", err)
+		return fmt.Errorf("error getting converted input: %w", err)
+	}
+
+	payload := processOutputData.Payload[2:]
+	if payload[2:10] == model.VOUCHER_SELECTOR {
```

Comment:
Não entendi porque esse if de novo

Reply:
1. [2:]     This slice notation means "starting from the index 2 up to the end of the slice." 
2. [2:10] This slice notation means "starting from the index 2 up to, but not including, index 10."